driver/src/main.rs

```diff
@@ -293,6 +274,38 @@ fn main() {
     scheduler.start();
 }
 
+fn setup_metrics() -> (StableXMetrics, HttpMetrics) {
+    let prometheus_registry = Arc::new(Registry::new());
+    let stablex_metrics = StableXMetrics::new(prometheus_registry.clone());
+    let http_metrics = HttpMetrics::new(&prometheus_registry).unwrap();
+    let metric_server = MetricsServer::new(prometheus_registry);
+    thread::spawn(move || {
+        metric_server.serve(9586);
+    });
+
+    (stablex_metrics, http_metrics)
+}
+
+fn setup_http_services(
+    http_factory: HttpFactory,
+    options: Options,
+) -> (Web3, GnosisSafeGasStation, PriceOracle) {
```

Comment:
Nit: To me it is cleaner to take `Options` by reference `&Options` and then clone the fields you do need owned inside of the function. That way we don't have to clone the whole thing. But this code is fine too so if you prefer this then feel free not to change it.

Reply:
Nah, I don't prefer and simply wasn't aware how to actually do this. Thanks for your meaningful and surely better suggested approach.